test/integration/controller/job/job_controller_test.go

```diff
@@ -63,7 +64,7 @@ var _ = ginkgo.Describe("Job controller", func() {
 		})
 		gomega.Expect(err).NotTo(gomega.HaveOccurred(), "failed to create manager")
 
-		err = workloadjob.NewReconciler(mgr.GetScheme(), mgr.GetClient()).SetupWithManager(mgr)
+		err = workloadjob.NewReconciler(mgr.GetScheme(), mgr.GetClient(), &record.FakeRecorder{}).SetupWithManager(mgr)
```

Comment:
we can do: `mgr.GetEventRecorderFor(constants.JobControllerName)`

if possible, can you check the events are produced throughout the test? not add a new test, just add a check that the event is produced, I think the test covers all cases where an event is supposed to be generated.

Reply:
sg, I will add these later.

Reply:
Added. I separate them into two commits.